src/vmm/src/builder.rs

```diff
@@ -338,27 +338,7 @@ pub fn build_microvm_for_boot(
     let initrd = load_initrd_from_config(boot_config, &guest_memory)?;
     // Clone the command-line so that a failed boot doesn't pollute the original.
     #[allow(unused_mut)]
-    let mut boot_cmdline = linux_loader::cmdline::Cmdline::new(arch::CMDLINE_MAX_SIZE);
-
-    // We're splitting the boot_args in regular parameters and init arguments.
-    // This is needed because on x86_64 we're altering the boot arguments by
-    // adding the virtio device configuration. We need to make sure that the init
-    // parameters are last, specified after -- as specified in the kernel docs
-    // (https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html).
-    let init_and_regular = boot_config
-        .cmdline
-        .as_str()
-        .split("--")
-        .collect::<Vec<&str>>();
-    if init_and_regular.len() > 2 {
-        return Err(StartMicrovmError::KernelCmdline(
-            "Too many `--` in kernel cmdline.".to_string(),
-        ));
-    }
-    let boot_args = init_and_regular[0];
-    let init_params = init_and_regular.get(1);
-
-    boot_cmdline.insert_str(boot_args)?;
+    let mut boot_cmdline = boot_config.cmdline.clone();
```

Comment:
I think we actually do need to keep it because there are still some points in the builder.rs file where we need to return a `StartMicrovmError` which is actually a `linux_loader::cmdline::Error` and it's automatically converted into a `StartMicrovmError::KernelCmdline` and propagated this way. For example, [here](https://github.com/firecracker-microvm/firecracker/blob/00ebfe26e97258b6f489ab38c47057a38aef7118/src/vmm/src/builder.rs#L968) we need to insert an argument and the insertion might fail; as a result, the error that is returned is a `linux_loader::cmdline::Error` which is converted into `StartMicrovmError::KernelCmdline` because the `std::convert::From<linux_loader::cmdline::Error> for StartMicrovmError` is implemented and we can propagate the error further.